### ISSUE TITLE
about the score_map

### ISSUE THREAD

**Issue by LiShenglana** [VISION-SJTU/USOT#15]
When I only use offline siamese module to train this model for a while, I find that the cls_score stop update and the score_map visualized like below:
![1670934545979](https://user-images.githubusercontent.com/106595830/207318323-2bc6ea78-4fca-4957-9225-bacb594bc2dd.png)
Do you know how can I fix this?
Thank you very much!

**Comment by zhengjilai** [comment]
@LiShenglana 
Hi ! Thanks for your attention on this work. 
What do you mean by "only using offline siamese module to train this model" ? In my experience, since there exists object scale/position shift on the training search areas, the response map should never become what you visualize above. 
Please give more details on your problem, e.g. your loss curve on the offline classification loss. Thank you!

**Comment by LiShenglana** [comment]
> @LiShenglana Hi ! Thanks for your attention on this work. What do you mean by "only using offline siamese module to train this model" ? In my experience, since there exists object scale/position shift on the training search areas, the response map should never become what you visualize above. Please give more details on your problem, e.g. your loss curve on the offline classification loss. Thank you!
This is the loss curve below:
![a173d45e4a3d5af1455e06e6089816d](https://user-images.githubusercontent.com/106595830/207516454-c47bce6b-1625-4e08-9d94-fd2cfac5512e.png)
![2692412eda6fad8b69fdc9e419c3271](https://user-images.githubusercontent.com/106595830/207516477-42931e50-e4b8-480d-8d00-83d8500c2bee.png)
At the begining, the score map looks like below:
![image](https://user-images.githubusercontent.com/106595830/207516833-0cf73782-4e53-4c08-be40-96ba681791b5.png)
But after 2 epoch, the score map looks like somthing wrong:
![image](https://user-images.githubusercontent.com/106595830/207516976-39a944a7-59be-4539-8549-5f1e5b232c46.png)
Thanks for your reply!

**Comment by zhengjilai** [comment]
@LiShenglana 

Bug definitely exists. Note that before training, the classification BCE loss is around 0.6. And after 2 epochs, your classification loss becomes around 0.4-0.5, as is shown in your loss curve. In my experience, a proper training process should lead to a classification loss gradually down to around 0.1, so what you met is absolutely abnormal. 

So I observed the training instances you illustrated, and found that **the classification labels seem to be wrong**. In other words, the spatial position of the template target in the search area does not have the corresponding positive label. Both two instances show similar phenomena. For example, in the training instance below, the target (a cyclist with its shadow) exactly lies in the center of the search area, but the corresponding positive area in the pseudo classification label (left) lies aound top-middle. This labeling error definitely **confuses the model**, and makes loss never decrease. 

I am sure that the original dataloader does not have such a labeling error, otherwise other followers of this repository should have already found it (TAT). Thus, I recommend you re-check your revisions on the codes (especially in the dataloader), or set some break points to debug the dataloader, in order to investigate why this labeling error happens.

**Comment by LiShenglana** [comment]
@zhengjilai Hi! I solved this problem following your advice. It is really the problem of dataloader that I revise before. Thank you very much!

**Comment by zhengjilai** [comment]
Glad to see the bug fixed.
